sniproxy.cpp

```diff
@@ -311,18 +305,38 @@ void SNIProxy::sendClick(uint8_t mouseButton, int x, int y)
     //note x,y are not actually where the mouse is, but the plasmoid
     //ideally we should make this match the plasmoid hit area
 
-    qCDebug(SNIPROXY) << "Sending click " << mouseButton << "to" << x << y;
+    qCDebug(SNIPROXY) << "Received click" << mouseButton << "with passed x*y" << x << y;
 
     auto c = QX11Info::connection();
 
-    //set our window so the middle is where the mouse is
+    auto cookie = xcb_query_pointer(c, m_windowId);
+    QScopedPointer<xcb_query_pointer_reply_t> pointer(xcb_query_pointer_reply(c, cookie, Q_NULLPTR));
+    /*qCDebug(SNIPROXY) << "samescreen" << pointer->same_screen << endl
+	<< "root x*y" << pointer->root_x << pointer->root_y << endl
+	<< "win x*y" << pointer->win_x << pointer->win_y;*/
+
+    //move our window so the mouse is within its geometry
+    uint32_t configVals[2] = {0, 0};
+    if (mouseButton >= XCB_BUTTON_INDEX_4) {
+	//scroll event, take pointer position
+	configVals[0] = pointer->root_x;
+	configVals[1] = pointer->root_y;
+    } else {
+	if (pointer->root_x > x + s_embedSize)
+	    configVals[0] = pointer->root_x - s_embedSize + 1;
```

Comment:
I don't understand. 

This positions the window so we're always nearly clicking the bottom right?